test/check-machines-create

```diff
@@ -2321,6 +2322,182 @@ vnc_password= "{vnc_passwd}"
+                                             location=config.VALID_DISK_IMAGE_PATH,
+                                             create_and_run=True,
+                                             delete=False)
+        vmCockpit = dialog.name
+        dialog.open() \
+            .fill()
+        b.click(".pf-v5-c-modal-box__footer button:contains(Create and run)")
+        self.browser.wait_not_present("#create-vm-dialog")
+        self.waitVmRow(vmCockpit, "system")
+        self.goToVmPage(vmCockpit)
+        b.wait_text(f"#vm-{vmCockpit}-system-state", "Running")
+        domainXML = m.execute(f"virsh dumpxml --inactive {vmCockpit}")
+
+        # on RHEL, SPICE is unsupported; we don't expect any device
+        if m.image.startswith("rhel") or m.image.startswith("centos"):
+            self.assertNotIn("spice", domainXML)
+            self.assertNotIn("qxl", domainXML)
+            # and thus no "Replace SPICE" menu entry
+            b.click(f"#vm-{vmCockpit}-system-action-kebab button")
+            b.wait_in_text(f"#vm-{vmCockpit}-system-action-kebab > .pf-v5-c-dropdown__menu", "Delete")
+            self.assertNotIn("Replace SPICE",
+                             b.text(f"#vm-{vmCockpit}-system-action-kebab > .pf-v5-c-dropdown__menu"))
+            return
+
+        # all other OSes configure SPICE by default
+        self.assertIn("graphics type='spice'", domainXML)
+        self.assertIn("channel type='spicevmc'", domainXML)
+        checkConnnectInfo(expect_vnc=True, expect_spice=True)
+
+        doReplaceSpice(vmCockpit, expect_running=True)
+
+        # "Replace SPICE" menu entry goes away (async)
+        for _retry in range(15):
+            time.sleep(1)
+            b.click(f"#vm-{vmCockpit}-system-action-kebab button")
+            b.wait_in_text(f"#vm-{vmCockpit}-system-action-kebab > .pf-v5-c-dropdown__menu", "Delete")
+            menu = b.text(f"#vm-{vmCockpit}-system-action-kebab > .pf-v5-c-dropdown__menu")
+            b.click(f"#vm-{vmCockpit}-system-action-kebab button")
+            b.wait_not_present(".pf-v5-c-dropdown__menu")
+            if "Replace SPICE" not in menu:
+                break
+        else:
+            self.fail("timed out waiting for Replace SPICE menu item removal")
+
+        # ensure that the VM can start without any SPICE plugins
+        with self.remove_spice_plugins():
+            # restart the VM to make the change effective
+            self.performAction(vmCockpit, "forceOff")
+            b.wait_not_present(f"#vm-{vmCockpit}-needs-shutdown")
+            b.click(f"#vm-{vmCockpit}-system-run")
+            b.wait_text(f"#vm-{vmCockpit}-system-state", "Running")
+            checkConnnectInfo(expect_vnc=True, expect_spice=False)
+
+        b.wait_not_present(f"#vm-{vmCockpit}-needs-shutdown")
+
+        self.goToMainPage()
+        self.machine.execute(f"virsh destroy {vmCockpit}; virsh undefine {vmCockpit}")
+
+        #
+        # custom spice-only VM, not running
```

Comment:
Can this part of the test run on RHEL?

I am a bit worried that we don't cover the actual use case: Replacing SPICE on RHEL in a VM that was created on an older version of that RHEL...

Reply:
Yes, let's do that! We only need to skip the "create with UI" part.